Sources/EDXMobileAnalytics/FullStory/FullStoryAnalyticsService.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  File.swift
+//  EDXMobileAnalytics
+//
+//  Created by Anton Yarmolenka on 20/02/2025.
+//
+
+import Foundation
+import FullStory
+import OEXFoundation
+import FirebaseCrashlytics
+
+public protocol CrashlyticsProtocol: Sendable {
+    func setCustomValue(_ value: Any?, forKey key: String)
+}
+extension Crashlytics: CrashlyticsProtocol, @unchecked @retroactive Sendable {}
+
+public protocol FullStoryProtocol {
+    static var delegate: FSDelegate? { get set }
+    static func restart()
+    static func identify(_ uid: String, userVars: [String: Any])
+    static func event(_ name: String, properties: [String: Any])
+    static func page(withName pageName: String, properties: [String: Any]?) -> any FSPage
+}
+extension FS: FullStoryProtocol {}
+
+final public class FullStoryAnalyticsService: NSObject, AnalyticsService, FSDelegate {
+    let firebaseEnabled: Bool
+    let crashlytics: CrashlyticsProtocol
+    let fullstoryType: FullStoryProtocol.Type
+    
+    public init(
+        _ firebaseEnabled: Bool,
+        crashlytics: CrashlyticsProtocol = Crashlytics.crashlytics(),
+        fullstoryType: FullStoryProtocol.Type = FS.self
+    ) {
```

Comment:
I think we should only keep `firebaseEnabled: Bool` parameter without `_` at the beginning. `CrashlyticsProtocol` and `FullStoryProtocol` should be made internal and removed from public initializer as they are only needed for testing purposes.

```suggestion
    public init(firebaseEnabled: Bool) { ... }
    
    init(
        firebaseEnabled: Bool,
        crashlytics: CrashlyticsProtocol = Crashlytics.crashlytics(),
        fullstoryType: FullStoryProtocol.Type = FS.self
    ) { ... }
```